component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -683,4 +713,54 @@ private boolean validateJTIInCache(String jti, SignedJWT signedJWT, JWTCacheEntr
         }
         return true;
     }
+
+    /**
+     * Validate whether the request signing algorithm is configured for the application.
+     *
+     * @param requestSigningAlgorithm     The request signed algorithm.
+     * @param clientId                    Client ID of the application.
+     * @return whether the request signing algorithm is configured for the application.
+     * @throws OAuthClientAuthnException OAuth Client Authentication Exception.
+     */
+    private boolean isValidSignatureAlgorithm(String requestSigningAlgorithm, String clientId)
+            throws OAuthClientAuthnException {
+
+        //   Obtain the signing algorithm configured for the application.
+        List<String> configuredSigningAlgorithms = getConfiguredSigningAlgorithm(clientId);
+        //  Validate whether the JWT signing algorithm is configured for the application.
+        if (configuredSigningAlgorithms == null || configuredSigningAlgorithms.contains(requestSigningAlgorithm)) {
```

Comment:
Need to check if the configuredSigningAlgorithms is empty as well. In case the service provider sends an empty list

Reply:
Fixed in e9d0e7eeb2260413659c89164eed65584a7d4278